### ISSUE TITLE
Component: Dialog (accessibility & focus issues)

### ISSUE THREAD

**Issue by Ivanho92** [primefaces/primeng#16997]
### Describe the bug

Primeng v18.0.0 offers a Dialog component (p-dialog) that does not meet accessibility requirements, as prescribed by [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role#required_javascript_features) and [W3C Web Accessibility Initiative](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/).

In summary, an accessible dialog should meet the following requirements:
1. When the dialog is shown in modal mode, the user should not be able to interact with other elements outside the dialog (focus trap)... 
2. ...and should not be able to keep scrolling on the page hidden by the dialog 
3. After opening the dialog, initial focus should be set on the first focusable element 
4. When the dialog closes, focus should be returned to the button that allowed opening the dialog 

Those requirements are met when there is at least one focusable element within the modal. ✔️
HOWEVER, If the modal does not contain any focusable element (e.g., only text), then those accessibility requirements are not met. ❌

### Environment

Chrome version 131.0.6778.109 (Official Build) (64-bit)

### Reproducer

https://stackblitz.com/edit/angular-primeng-18-accessible-dialog

### Angular version

18.2.13

### PrimeNG version

18.0.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

v18.20.3

### Browser(s)

Chrome 131

### Steps to reproduce the behavior

1. go to https://stackblitz.com/edit/angular-primeng-18-accessible-dialog
2. navigate with keyboard (press "tab") to button "Show Form Dialog" and press enter
    - dialog should display and first input element ("Username") should be focused ✔️
    - accessibility requirements 1, 2 and 3 mentionned above should be met ✔️
3. press "escape" to hide dialog
    - accessibility requirement 4 is not met: focus is not returned to button "Show Form Dialog" ❌
5. navigate with keyboard to button "Show Text Dialog" and press enter
    - dialog should display and first input element ("close" button, X mark icon) should be focused ❌
    - accessibility requirements 1, 2, 3 are not met: the user can still navigate and interact with other elements through keyboard navigation ❌

### Expected behavior

- all accessibility requirements should be met for ALL dialogs, including the ones that do contain only text
- after closing the dialog, the focus should be returned to focusable element that triggered the dialog opening
